File: ProjetoFase02.cs
using System;

public class Banco
{
    private string _nome;
    private string _codigoBACEN;

    public string Nome
    {
        get { return _nome.ToUpper(); }
        set
        {
            if (!string.IsNullOrWhiteSpace(value))
                _nome = value;
        }
    }

    public string CodigoBACEN
    {
        get { return _codigoBACEN; }
        set
        {
            if (!string.IsNullOrWhiteSpace(value))
                _codigoBACEN = value;
        }
    }
}

public class Transacao
{
    private decimal _valor;

    public decimal Valor
    {
        get { return _valor; }
        set
        {
            if (value > 0)
                _valor = value;
        }
    }

    public DateTime Data { get; set; }
    public string ContaOrigem { get; set; }
    public string ContaDestino { get; set; }
}

class Program
{
    static void Main()
    {
        Banco banco = new Banco();
        banco.Nome = "Banco do Brasil";
        banco.CodigoBACEN = "001";

        Transacao t = new Transacao();
        t.Valor = 150;
        t.Data = DateTime.Now;
        t.ContaOrigem = "Conta A";
        t.ContaDestino = "Conta B";

        Console.WriteLine("Banco: " + banco.Nome);
        Console.WriteLine("Código BACEN: " + banco.CodigoBACEN);
        Console.WriteLine("Valor da Transação: " + t.Valor);
    }
}

File: ProjetoFase02.py
class Banco:
    def __init__(self, nome, codigo_bacen):
        self.nome = nome
        self.codigo_bacen = codigo_bacen

    def set_nome(self, nome):
        if nome.strip() != "":
            self.nome = nome

    def get_nome(self):
        return self.nome.upper()

    def set_codigo_bacen(self, codigo):
        if codigo.strip() != "":
            self.codigo_bacen = codigo

    def get_codigo_bacen(self):
        return self.codigo_bacen


class Transacao:
    def __init__(self, valor, data, conta_origem, conta_destino):
        self.valor = valor
        self.data = data
        self.conta_origem = conta_origem
        self.conta_destino = conta_destino

    def set_valor(self, valor):
        if valor > 0:
            self.valor = valor

    def get_valor(self):
        return self.valor


# Main
banco = Banco("Banco do Brasil", "001")
transacao = Transacao(150, "20/02/2026", "Conta A", "Conta B")

print("Banco:", banco.get_nome())
print("Código BACEN:", banco.get_codigo_bacen())
print("Valor da Transação:", transacao.get_valor())
